packages/server/tests/ListUserFollowers.test.ts

```diff
@@ -0,0 +1,385 @@
+                tmpAuthUserID: requestingUserID,
+            } as ListUserFollowersRequestBody)
+            .expect('Content-Type', /json/)
+            .expect(200);
+        const page2BodyParsed =
+            ListUserFollowersResponseBody.parse(page2BodyRaw);
+        console.log({ pageBodyParsed: page2BodyParsed });
+        assert.equal(page2BodyParsed.page, 2);
+        assert.equal(
+            page2BodyParsed.totalEntries,
+            sortedByNicknameSearchedUserFollowersUserSummary.length,
+        );
+        assert.isFalse(page2BodyParsed.hasMore);
+        assert.equal(
+            page2BodyParsed.data.length,
+            sortedByNicknameSearchedUserFollowersUserSummary.length -
+                PAGE_MAX_LENGTH,
+        );
+        assert.deepEqual(
+            page2BodyParsed.data,
+            sortedByNicknameSearchedUserFollowersUserSummary.slice(
+                PAGE_MAX_LENGTH,
+                PAGE_MAX_LENGTH * 2,
+            ),
+        );
+    });
+
+    test('It should retrieve filtered user followers', async (assert) => {
+        const requestingUserID = datatype.uuid();
+        const searchedUserID = datatype.uuid();
+        const searchedUser = await User.create({
+            uuid: searchedUserID,
+            nickname: internet.userName(),
+        });
+        await User.create({
+            uuid: requestingUserID,
+            nickname: internet.userName(),
+        });
+
+        const searchQuery = datatype.string(1);
+        const users = await User.createMany(
+            generateArray({
+                fill: (index) => ({
+                    uuid: datatype.uuid(),
+                    nickname: `${
+                        index % 2 === 0 ? searchQuery : ''
+                    }${internet.userName()}`,
+                }),
+                minLength: 20,
+                maxLength: 20,
+            }),
+        );
+
+        //Follow relationship set up
+        const followersCounter = datatype.number({
+            max: 10,
+            min: 8,
+        });
+        const searchedUserFollowersUserSummary: UserSummary[] = (
+            await Promise.all(
+                users.map(async (user, index) => {
+                    if (index < followersCounter) {
+                        await searchedUser.related('followers').save(user);
+                        return {
+                            userID: user.uuid,
+                            nickname: user.nickname,
+                        };
+                    }
+
+                    return undefined;
+                }),
+            )
+        ).filter(
+            (el: UserSummary | undefined): el is UserSummary =>
+                el !== undefined,
+        );
+        ///
+
+        //filtering options
+        const matchingSearchQuerySortedByNicknameSearchedUserFollowers =
+            searchedUserFollowersUserSummary
+                .filter((user) => user.nickname.startsWith(searchQuery))
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏‏In the database we filter users with `ilike` operator. I think that the `filter()` method here should be case-insensitive too.

Reply:
Indeed you're right, but isn't it not required as I manually prepend `searchQuery ` at the nicknames beginning ? :thinking:

Reply:
So if I understand correctly, you would see something either like:
```ts
        //filtering options
        const matchingSearchQuerySortedByNicknameSearchedUserFollowers =
            searchedUserFollowersUserSummary
                .filter((user) =>
                    user.nickname.toLowerCase().startsWith(searchQuery),
                )
                .sort((a, b) => {
                    if (a.nickname.toLowerCase() < b.nickname.toLowerCase()) {
                        return -1;
                    }
                    if (a.nickname.toLowerCase() > b.nickname.toLowerCase()) {
                        return 1;
                    }
                    return 0;
                });
        ///
```

nor

```ts
        //filtering options
        const matchingSearchQuerySearchedUserFollowersUserSummary =
            searchedUserFollowersUserSummary.filter((user) =>
                user.nickname.toLowerCase().startsWith(searchQuery),
            );
        const matchingSearchQuerySortedByNicknameSearchedUserFollowers =
            matchingSearchQuerySearchedUserFollowersUserSummary.sort((a, b) => {
                if (a.nickname.toLowerCase() < b.nickname.toLowerCase()) {
                    return -1;
                }
                if (a.nickname.toLowerCase() > b.nickname.toLowerCase()) {
                    return 1;
                }
                return 0;
            });
        ///
```

Reply:
Something like that:

```ts
        const matchingSearchQuerySortedByNicknameSearchedUserFollowers =
            searchedUserFollowersUserSummary
                .filter((user) =>
                    // both `user.nickname` and `searchQuery` should be lower-cased
                    user.nickname.toLowerCase().startsWith(searchQuery.toLowerCase()),
                )
                .sort((a, b) => {
                    if (a.nickname.toLowerCase() < b.nickname.toLowerCase()) {
                        return -1;
                    }
                    if (a.nickname.toLowerCase() > b.nickname.toLowerCase()) {
                        return 1;
                    }
                    return 0;
                });
```

Reply:
Ok I see makes sense !
Pushing an update

Reply:
Wondering if on mac OS those tests would fail :open_mouth: 
Could you run them? ( [related article](https://dba.stackexchange.com/questions/106964/why-is-my-postgresql-order-by-case-insensitive/131471#131471) )

Reply:
I'll try now. I think it won't fail as our PostgreSQL database is run inside a Docker container, which is presumably based on Linux 😮.
Interesting question!

Reply:
The tests succeeded. 👌